include/dca/function/function.hpp

```diff
@@ -109,7 +114,11 @@ class function {
     return Nb_sbdms;
   }
   std::size_t size() const {
-    return Nb_elements;
+    return nb_elements_;
+  }
+
+  void resize(std::size_t nb_elements_new) {
+    nb_elements_ = nb_elements_new;
```

Comment:
what is the point of this function? this looks fishy.

Reply:
In tp_accumulator_gpu.hpp, we default allocate G4 with full size on CPU from base class https://github.com/CompFUSE/DCA/blob/master/include/dca/phys/dca_step/cluster_solver/shared_tools/accumulation/tp/tp_accumulator_gpu.hpp#L164 and copy G4 vector from GPU to CPU here: `https://github.com/CompFUSE/DCA/blob/master/include/dca/phys/dca_step/cluster_solver/shared_tools/accumulation/tp/tp_accumulator_gpu.hpp#L453`. If the size of source and destination is not matched, an error will occur. 

In distributed G4 case, G4 size from both GPU and CPU size are both reduced to 1/ranks. However, the function class does not support templated initialization with distributed G4 parameters. So I added a helper function to function class to resize function size.

Reply:
It wasn't an elegant solution though. However, in DcaData class, G4 has been already allocated and initialized on CPU in here: https://github.com/STEllAR-GROUP/DCA/blob/distG4_pr/include/dca/phys/dca_data/dca_data.hpp#L308-L313. Is it possible to replace BaseClass::G4_ with DcaData::G4_?